### DISCUSSION TITLE
Control Interface

### DISCUSSION THREAD

**Discussion by MichaelM4** [keirf/flashfloppy#551] [General]
Hello,

Is there a communication port that can be used to control FlashFloppy?

For example a serial port that can be used to get the USB drive contents and mount a floppy image.

Regards,
Mike

**Comment by GrahamClemo** [comment]
I'd be interested in this too, as I'd like to connect an ESP32/8266 to a spare set of UART pins on the Gotek, and then connect to it via a web or mobile interface.

**Comment by hcsaustrup** [comment]
This could be very interesting - I've got a CP/M system with a couple of Goteks running FlashFloppy in the basement, and I'd love to somehow be able to switch disks from the comfort of my office. Either from the system itself, or through some other device like a RaspberryPi. Clearly I can trigger the buttons, but I will be unaware of exactly what image is currently loaded. A very basic interface on a serial console would be awesome.

**Reply by muskeldata** [reply]
Hi! I'm currently trying to install a Gotek with flashfloppy on a CP/M system (Telenova Compis) with no luck so far. I'd love to hear what system you're running this on and how you configured it.

**Comment by hcsaustrup** [comment]
Would it perhaps be possible to use the RaspberryPi's I2C interface to talk to FlashFloppy using the [FlashFloppy-OSD](https://github.com/keirf/flashfloppy-osd) interface? That would be pretty neat! :-)

**Comment by GrahamClemo** [comment]
I think there's already UART pins on at least some Gotek devices which can be used for debugging output, so it may be possible to repurpose those as a remote control.

**Comment by twistedsymphony** [comment]
I have a project that needs such an interface. it's old arcade hardware and I have an ESP32 that is setting hardware configuration on the hardware but I also need to select the appropriate disc image. would be great if I could use the ESP32 to also select the corresponding disc image.
